sound/hda/intel-dsp-config.c

```diff
@@ -558,8 +558,29 @@ static const struct config_entry *snd_intel_dsp_find_config
 			int i;
 
 			for (i = 0; i < table->codec_hid->num_codecs; i++)
-				if (acpi_dev_present(table->codec_hid->codecs[i], NULL, -1))
-					break;
+				if (acpi_dev_present(table->codec_hid->codecs[i], NULL, -1)) {
+					struct nhlt_acpi_table *nhlt;
+					bool ssp_found = false;
+
+					nhlt = intel_nhlt_init(&pci->dev);
+					if (!nhlt) {
+						dev_err(&pci->dev, "%s: NHLT table not found, skipped HID %s\n",
+						 __func__, table->codec_hid->codecs[i]);
```

Comment:
Would it be better to use dev_warn like line 581? Not sure if missing NHLT table is an error or not.

Reply:
I think it's an error because in that case the Windows drivers would not work at all, so that's really a broken platform. NHLT needs to be present/valid to support I2S codecs.

The dev_warn is different, essentially there's a table that's not needed because the hardware is NOT I2S based.

But if there's consensus that dev_warn is better I don't mind scaling dev_err back. It's a corner case anyways.

Reply:
`dev_err()` when showed in normal logs (first of all the log level can be adjusted) might have a different colour, hence attention by the user. I think since we use `continue;`, meaning non-fatal, `dev_warn()` less scares the end user.

Reply:
dev_warn() is is then, thanks for the feedback.